WordPress/gutenberg
# [core-data] Document and add types for dynamic actions and selectors.

Fixes #56287
Related #67142

## What?
Define the type declarations for dynamic selectors and actions for `core-data` package`

## Why?

The dynamic selectors cause errors when used in TypeScript because TS can't find those defined anywhere.

## How?

The PR creates TS definition declarations for all the dynamic selectors and some useful actions

## Testing Instructions
- Use the dynamic selectors or actions in TypeScript files
- Confirm that TS doesn't show an error

Example

```ts
import { store as coreStore } from '@wordpress/core-data';
import { select } from '@wordpress/data';

const { getUser, getUsers } = select( coreStore );
```

### Testing Instructions for Keyboard


## Screenshots or screencast 


<img width="1126" alt="image" src="https://github.com/user-attachments/assets/2912bb5a-eea2-4277-be80-c9cfe285bb0f">

We will now have auto-completions for entity selectors
<img width="1192" alt="image" src="https://github.com/user-attachments/assets/c29cff02-0ea9-45db-b7ff-5699193a6131">
